pkg/aws/eni/node.go

```diff
@@ -279,8 +288,21 @@ func (n *Node) allocateENI(s *types.Subnet, a *allocatableResources) error {
 	scopedLog = scopedLog.WithField(fieldEniID, eniID)
 	scopedLog.Info("Created new ENI")
 
+	attachRetries := 0
+retryAttachment:
```

Comment:
I personally think that 
```
	for attachRetries < maxAttachRetries {
		attachmentID, err := n.manager.ec2API.AttachNetworkInterface(index, nodeResource.Spec.ENI.InstanceID, eniID)
		if err != nil {
			//(...)
			attachRetries++
			index = n.findNextIndex(index + 1)
		} else {
			break
		}
	}
```
Would work better here, but it's my personal preference against using `goto`.